### PR TITLE
Consolidating initializing message to a single line

### PR DESCRIPTION
**Pull request by AkashGutha**
### Requirements

Consolidating the logging info into a single line.

### Description of the Change

Still a work in progress. So, the idea is to implement a combined logger.
is that the aim of #203

### Alternate Designs

If not we can just remove one of the log lines

### Benefits

Having a combined logger will make things simpler going forward.

### Possible Drawbacks

I possibly don't see any.

### Applicable Issues

#203

### CODE REVIEW

**Comment by StanleyGoldman** [comment]
Hey @AkashGutha thanks for participating! This task is a really simple one.
I'll leave some comments to give you some direction...

**Comment by StanleyGoldman** [comment]
Okay, I understand why you might be confused, you are looking at two different types of loggers there...

`Debug.Log` is the standard Unity console log.
`Logging.Info` is our custom logger that writes to the log file on disk.

For #203 we are looking to combine the messages we send to the Unity console log.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs [hunk 1]
```diff
@@ -37,6 +37,7 @@ private static void Initialize()
 
             if (ApplicationCache.Instance.FirstRun)
             {
+                // TODO: should be replaced with a common logger ?
                 Debug.Log("Initialized GitHub for Unity version " + ApplicationInfo.Version);
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs]: Combine the message we send to this `Debug.Log`..

File: src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs [hunk 2]
```diff
@@ -52,10 +53,11 @@ private static void Initialize()
                 {
                     Logging.Error(ex, "Error rotating log files");
                 }
-
+                // TODO: should be replaced with a common logger ?
                 Debug.Log("Initialized GitHub for Unity log file: " + logPath);
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs]: ... with the message we send to this `Debug.Log`

File: src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs [hunk 3]
```diff
@@ -52,10 +53,11 @@ private static void Initialize()
...
             }
             Logging.LogAdapter = new FileLogAdapter(logPath);
+            // TODO: should be replaced with a common logger ?
             Logging.Info("Initializing GitHub for Unity version " + ApplicationInfo.Version);
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs]: This one should be left alone.

**Comment by AkashGutha** [comment]
Also, I wanted to raise a discussion.

Why not use interpolated strings for the entire project?

**Comment by StanleyGoldman** [comment]
### TLDR;
`GitHub.Unity` is restricted to C# 4 language features. You cannot use string interpolation, `async`/`await` and other C# 6.0 goodies in `GitHub.Unity`.

### Long Form Answer
This is a part of how our project and solution is configured and that is a very good question...
I want you to notice two things...

- Language Configuration
  - `GitHub.*` is set to c# 6 language features
  - `GitHub.Unity` is restricted to C# 4 language features
- Folder Structure
  - `GitHub.Api` and `GitHub.Logging` can be found in `/src/`
  - `GitHub.Unity` can be found in `/src/UnityExtension/Assets/Editor`

This allows us to open `/src/UnityExtension` as a project in Unity.
We mostly do this to quickly prototype UI changes at run time in Unity.

**Comment by StanleyGoldman** [comment]
FYI we are talking about these two messages

img

**Comment by AkashGutha** [comment]
I'm a bit curious as to what would be the default log message if it wasn't the firsts run?

**Comment by StanleyGoldman** [comment]
That `FirstRun` section only runs when you first open Unity.
If you had that code outside that `if` statement, it would run every time you pushed play and stop in Unity.

**Comment by AkashGutha** [comment]
I guess that should be fixed. 
Having troubles setting up build. 
I'll post once done.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs [hunk 4]
```diff
@@ -53,10 +52,15 @@ private static void Initialize()
...
+                logInfo = "Initialized GitHub for Unity verison " + ApplicationInfo.Version + " and Unity log file: " + logPath;
+                Debug.Log(logInfo);
             }
+
             Logging.LogAdapter = new FileLogAdapter(logPath);
-            Logging.Info("Initializing GitHub for Unity version " + ApplicationInfo.Version);
+            if (logInfo != "")
+            {
+                Logging.Info(logInfo);
+            }
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs]: Please revert this chunk. We were only looking to change the log message above.
